app/src/main/java/com/example/advanceduiapp/ui/pager/PagerFragment.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.advanceduiapp.ui.pager
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import com.example.advanceduiapp.databinding.FragmentPagerBinding
+
+class PagerFragment(private val imageResource: Int, private val title: String, private val showButton: Boolean) : Fragment() {
+
+    private var _binding: FragmentPagerBinding? = null
+    private val model: RecyclerViewModel by activityViewModels()
```

Comment:
viewModel should have name similar to the fragment that it is related to -> PagerFragment + PagerViewModel